source/shared/cpp/ObjectModel/ParseUtil.h

```diff
@@ -103,11 +103,11 @@ namespace AdaptiveSharedNamespace
         std::string ToLowercase(const std::string& value);
     };
 
-    template<typename T>
+    template<typename T, typename Fn>
```

Comment:
>typename [](start = 25, length = 8)

Hey @chriche-ms  are there known perf issues with std::function that are prompting this change? Quick search seems to indicate so ? Also I'm wondering if you'd be up for a quick "Perf gotchas" type of learnings for us for education as you've been wrking on profiling the library here -- are there specific best practices resources you've been looking at as guiding your analysis here? It might be a great topic to visit all up say in our next sync maybe - the worst offenders as far as perf is concerned based on your profiling and case study?

Reply:
Yep, there definitely can be performance concerns with using std::function although in this case the change was more to do with it being used incorrectly. Unlike C# delegates, std::function is mostly for storing (and calling obviously!) callable types (e.g. used as a member variable) and not meant to be used as a function parameter. Where possible templated parameters should be used for passing callable types.